servers/Fabric.Mcp.Server/vscode/src/extension.ts

```diff
@@ -0,0 +1,148 @@
+    const arch = process.arch;
+    if (process.platform === 'win32') {
+        if (arch === 'x64' || arch === 'arm64') {
+            binary = 'azmcp.exe';
+        } else {
+            throw new Error('Unsupported Windows architecture: ' + arch);
+        }
+    } else if (process.platform === 'darwin' || process.platform === 'linux') {
+        if (arch === 'x64' || arch === 'arm64') {
+            binary = 'azmcp';
+        } else {
+            throw new Error(`Unsupported ${process.platform} architecture: ${arch}`);
+        }
+    } else {
+        throw new Error('Unsupported platform: ' + process.platform);
+    }
+
+    // Use the binary from the extension's server/os folder
+    const binPath = path.join(context.extensionPath, 'server', binary);
+    if (!fs.existsSync(binPath)) {
+        throw new Error(`azmcp binary not found at ${binPath}. Please ensure the server binary is present.`);
+    }
+
+    // Ensure executable permission on macOS and Linux (once at activation)
+    if ((process.platform === 'linux' || process.platform === 'darwin') && fs.existsSync(binPath)) {
+        try {
+            fs.chmodSync(binPath, 0o755);
+        } catch (e) {
+            console.warn(`Failed to set executable permission on ${binPath}: ${e}`);
+        }
+    }
+
+    context.subscriptions.push(
+        vscode.lm.registerMcpServerDefinitionProvider('fabricMcpProvider', {
+            onDidChangeMcpServerDefinitions: didChangeEmitter.event,
+            provideMcpServerDefinitions: async () => {
+                // Read enabled MCP services from user/workspace settings
+                const config = vscode.workspace.getConfiguration('fabricMcp');
+                // Example: ["storage", "keyvault", ...]
+                const enabledServices: string[] | undefined = config.get('enabledServices');
+                const args = ['server', 'start'];
+
+                // Server Mode (single | namespace | all). Default 'namespace'.
+                const mode = config.get<string>('serverMode') || 'namespace';
+                if (mode) {
+                    args.push('--mode', mode);
+                }
+
+                // Namespaces filter
+                if (enabledServices && Array.isArray(enabledServices) && enabledServices.length > 0) {
+                    for (const svc of enabledServices) {
+                        args.push('--namespace', svc);
+                    }
+                }
+
+                // Read-only flag
+                const readOnly = config.get<boolean>('readOnly') === true;
+                if (readOnly) {
+                    args.push('--read-only');
+                }
+
+
+                // Honor VS Code telemetry settings
+                // Only set FABRIC_MCP_COLLECT_TELEMETRY if telemetry is disabled
+                const env: Record<string, string | number | null> = {};
+                if (!vscode.env.isTelemetryEnabled) {
+                    env.FABRIC_MCP_COLLECT_TELEMETRY = 'false';
```

Comment:
@conniey , is this env variable correct here ?

Reply:
Will implement Fabric in next pr